web/frontend/src/components/navigation/components/nav.svelte

```diff
@@ -23,54 +27,165 @@ const handleRemoveWaypoint = async (id: string) => {
   }
 };
 
+const handleAddObstacle = () => {
```

Comment:
This looks like a fair amount of non-website link navigation (i.e. `<nav>`) concerns  mixing with robot navigation concerns. Should this component be split up for testability?

Reply:
Yeah, I think this makes sense, will split this up a bit